File: test-sdk-params.py
"""
Test script to verify Python SDK tools accept and pass language/organization parameters
"""

import asyncio
from translation_helps import TranslationHelpsClient


async def test_sdk_parameters():
    print('🧪 Testing Python SDK Parameter Passing\n')

    client = TranslationHelpsClient({
        'serverUrl': 'https://tc-helps.mcp.servant.bible/api/mcp',
    })

    try:
        await client.connect()
        print('✅ Connected to MCP server\n')

        # Test 1: Default parameters (should use "en" and "unfoldingWord")
        print('Test 1: Using default parameters (should use "en" and "unfoldingWord")')
        try:
            result1 = await client.fetch_scripture({
                'reference': 'John 3:16',
                # No language or organization specified
            })
            print('✅ Default params test passed')
            print(f'   Response length: {len(result1)} chars\n')
        except Exception as e:
            print(f'❌ Default params test failed: {str(e)}\n')

        # Test 2: Custom language parameter
        print('Test 2: Using custom language parameter (language="es")')
        try:
            result2 = await client.fetch_scripture({
                'reference': 'John 3:16',
                'language': 'es',
                # No organization specified (should default to "unfoldingWord")
            })
            print('✅ Custom language test passed')
            print(f'   Response length: {len(result2)} chars\n')
        except Exception as e:
            print(f'❌ Custom language test failed: {str(e)}\n')

        # Test 3: Custom organization parameter
        print('Test 3: Using custom organization parameter (organization="testOrg")')
        try:
            result3 = await client.fetch_scripture({
                'reference': 'John 3:16',
                'organization': 'testOrg',
                # No language specified (should default to "en")
            })
            print('✅ Custom organization test passed')
            print(f'   Response length: {len(result3)} chars\n')
        except Exception as e:
            print(f'❌ Custom organization test failed: {str(e)}\n')

        # Test 4: Both custom parameters
        print('Test 4: Using both custom parameters (language="es", organization="testOrg")')
        try:
            result4 = await client.fetch_scripture({
                'reference': 'John 3:16',
                'language': 'es',
                'organization': 'testOrg',
            })
            print('✅ Both custom params test passed')
            print(f'   Response length: {len(result4)} chars\n')
        except Exception as e:
            print(f'❌ Both custom params test failed: {str(e)}\n')

        # Test 5: Test with fetch_translation_notes
        print('Test 5: Testing fetch_translation_notes with custom parameters')
        try:
            result5 = await client.fetch_translation_notes({
                'reference': 'John 3:16',
                'language': 'es',
                'organization': 'unfoldingWord',
            })
            print('✅ fetch_translation_notes test passed')
            print(f'   Response type: {type(result5)}\n')
        except Exception as e:
            print(f'❌ fetch_translation_notes test failed: {str(e)}\n')

        # Test 6: Test with fetch_translation_word
        print('Test 6: Testing fetch_translation_word with custom parameters')
        try:
            result6 = await client.fetch_translation_word({
                'term': 'love',
                'language': 'es',
                'organization': 'unfoldingWord',
            })
            print('✅ fetch_translation_word test passed')
            print(f'   Response type: {type(result6)}\n')
        except Exception as e:
            print(f'❌ fetch_translation_word test failed: {str(e)}\n')

        print('✅ All tests completed!')
    except Exception as e:
        print(f'❌ Test failed: {str(e)}')
        raise


if __name__ == '__main__':
    asyncio.run(test_sdk_parameters())

File: test-sdk-params.js
/**
 * Test script to verify SDK tools accept and pass language/organization parameters
 */

import { TranslationHelpsClient } from "@translation-helps/mcp-client";

async function testSDKParameters() {
  console.log("🧪 Testing SDK Parameter Passing\n");

  const client = new TranslationHelpsClient({
    serverUrl: "https://tc-helps.mcp.servant.bible/api/mcp",
  });

  try {
    await client.connect();
    console.log("✅ Connected to MCP server\n");

    // Test 1: Default parameters (should use "en" and "unfoldingWord")
    console.log(
      'Test 1: Using default parameters (should use "en" and "unfoldingWord")',
    );
    try {
      const result1 = await client.fetchScripture({
        reference: "John 3:16",
        // No language or organization specified
      });
      console.log("✅ Default params test passed");
      console.log(`   Response length: ${result1.length} chars\n`);
    } catch (error) {
      console.log("❌ Default params test failed:", error.message);
    }

    // Test 2: Custom language parameter
    console.log('Test 2: Using custom language parameter (language="es")');
    try {
      const result2 = await client.fetchScripture({
        reference: "John 3:16",
        language: "es",
        // No organization specified (should default to "unfoldingWord")
      });
      console.log("✅ Custom language test passed");
      console.log(`   Response length: ${result2.length} chars\n`);
    } catch (error) {
      console.log("❌ Custom language test failed:", error.message);
    }

    // Test 3: Custom organization parameter
    console.log(
      'Test 3: Using custom organization parameter (organization="testOrg")',
    );
    try {
      const result3 = await client.fetchScripture({
        reference: "John 3:16",
        organization: "testOrg",
        // No language specified (should default to "en")
      });
      console.log("✅ Custom organization test passed");
      console.log(`   Response length: ${result3.length} chars\n`);
    } catch (error) {
      console.log("❌ Custom organization test failed:", error.message);
    }

    // Test 4: Both custom parameters
    console.log(
      'Test 4: Using both custom parameters (language="es", organization="testOrg")',
    );
    try {
      const result4 = await client.fetchScripture({
        reference: "John 3:16",
        language: "es",
        organization: "testOrg",
      });
      console.log("✅ Both custom params test passed");
      console.log(`   Response length: ${result4.length} chars\n`);
    } catch (error) {
      console.log("❌ Both custom params test failed:", error.message);
    }

    // Test 5: Test with fetchTranslationNotes
    console.log("Test 5: Testing fetchTranslationNotes with custom parameters");
    try {
      const result5 = await client.fetchTranslationNotes({
        reference: "John 3:16",
        language: "es",
        organization: "unfoldingWord",
      });
      console.log("✅ fetchTranslationNotes test passed");
      console.log(`   Response type: ${typeof result5}\n`);
    } catch (error) {
      console.log("❌ fetchTranslationNotes test failed:", error.message);
    }

    // Test 6: Test with fetchTranslationWord
    console.log("Test 6: Testing fetchTranslationWord with custom parameters");
    try {
      const result6 = await client.fetchTranslationWord({
        term: "love",
        language: "es",
        organization: "unfoldingWord",
      });
      console.log("✅ fetchTranslationWord test passed");
      console.log(`   Response type: ${typeof result6}\n`);
    } catch (error) {
      console.log("❌ fetchTranslationWord test failed:", error.message);
    }

    console.log("✅ All tests completed!");
  } catch (error) {
    console.error("❌ Test failed:", error);
    process.exit(1);
  }
}

testSDKParameters();
